internal/clusterfeature/features/logging/operator.go

```diff
@@ -368,20 +374,108 @@ func (op FeatureOperator) processLoki(ctx context.Context, spec lokiSpec, cl clu
 			return errors.WrapIf(err, "failed to merge loki values with config")
 		}
 
-		return op.helmService.ApplyDeployment(
+		if err := op.helmService.ApplyDeployment(
 			ctx,
 			cl.GetID(),
 			op.config.Namespace,
 			chartName,
 			lokiReleaseName,
 			valuesBytes,
 			chartVersion,
-		)
+		); err != nil {
+			return errors.WrapIf(err, "failed to apply Loki deployment")
+		}
+
+		if err := op.createLokiOutputDefinition(ctx, cl); err != nil {
+			return errors.WrapIf(err, "failed to create output definition for Loki")
+		}
+
+		if err := op.createLokiFlowResource(ctx, cl.GetID()); err != nil {
+			return errors.WrapIf(err, "failed to create flow resource for Loki")
+		}
 	}
 
 	return nil
 }
 
+func (op FeatureOperator) createLokiOutputDefinition(ctx context.Context, cl clusterfeatureadapter.Cluster) error {
+	k8sConfig, err := cl.GetK8sConfig()
+	if err != nil {
+		return errors.WrapIfWithDetails(err, "failed to get kubeconfig", "cluster", cl.GetID())
+	}
+
+	serviceURL, err := op.endpointsService.GetServiceURL(k8sConfig, lokiServiceName, op.config.Namespace)
+	if err != nil {
+		return errors.WrapIf(err, "failed to get Loki service url")
+	}
+
+	// delete former Loki outputs
+	var formerOutputs v1beta1.ClusterOutputList
+	if err := op.kubernetesService.List(ctx, cl.GetID(), &formerOutputs); err != nil {
+		return errors.WrapIf(err, "failed to list cluster outputs")
+	}
+	for _, item := range formerOutputs.Items {
+		if item.Name == lokiOutputDefinitionName {
+			if err := op.kubernetesService.DeleteObject(ctx, cl.GetID(), &item); err != nil {
+				return errors.WrapIf(err, "failed to delete Loki cluster outputs")
+			}
+		}
+	}
+
+	var outputDef = &v1beta1.ClusterOutput{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       loggingOperatorKindClusterOutput,
+			APIVersion: loggingOperatorAPIVersion,
+		},
```

Comment:
nitpick: I think you can omit this, although it shouldn't hurt

Reply:
thx @pepov , i removed it